src/components/Router/index.js

```diff
@@ -447,35 +447,19 @@ class HistoryRouter extends Component {
   }
 
   formattedError = (response, status) => {
-    if (typeof response === 'string') {
-      // TODO: this if statement should be deleted once all APIs start using the same signature for responses
-      // Currently detect_documents returns just a string. Examples:
-      // `Could not decode token: hello`
-      // `Token has expired.`
-      // Telephony returns the a JSON response
-      // {“unauthorized”:”Could not decode token: hello"}
-      // {"unauthorized":"Token has expired."}
-      // Tickets in backlog to update all APIs to use signature similar to main Onfido API
-      let message
-      try {
-        const jsonRes = JSON.parse(response)
-        message = jsonRes.unauthorized || jsonRes.error || response
-      } catch {
-        // response is just a string so we will return it as the message
-        message = response
-      }
-      const type = message.includes('expired') ? 'expired_token' : 'exception'
-      return { type, message }
-    }
-    const apiError = response.error || {}
+    const errorResponse = response.error || response || {}
+    // TODO: remove once for find_document_in_image back-end for `/validate_document` returns error response with same signature
+    const isDocAutoCaptureError =
+      typeof response === 'string' && response.includes('expired')
     const isExpiredTokenError =
-      status === 401 && apiError.type === 'expired_token'
+      status === 401 &&
+      (isDocAutoCaptureError || errorResponse.type === 'expired_token')
     const type = isExpiredTokenError ? 'expired_token' : 'exception'
     // TODO: delete response.reason once `v2/live_video_challenge` endpoints starts using the same signature for responses
     // `v2/live_video_challenge` returns a generic message for both invalid and expired tokens. Example:
     // {"reason":"invalid_token","status":"error"}
     // Ticket in backlog to update all APIs to use signature similar to main Onfido API
-    const message = apiError.message || response.reason
+    const message = errorResponse.message || response.reason
     return { type, message }
```

Comment:
If `response` is a `string`, do you want this to return `{ type: 'expired_token', message: undefined }`?

Reply:
Good point. The message itself will only say "Token expired" but that's still better than nothing.